temporal-sdk/src/main/java/io/temporal/internal/client/RootWorkflowClientInvoker.java

```diff
@@ -334,8 +334,17 @@ public <R> StartUpdateOutput<R> startUpdate(StartUpdateInput<R> input) {
             .setRequest(request)
             .build();
     Deadline pollTimeoutDeadline = Deadline.after(POLL_UPDATE_TIMEOUT_S, TimeUnit.SECONDS);
```

Comment:
Shouldn't the deadline be in the loop?

Reply:
Arguably it doesn't need to be set at all

Reply:
I unset this in the most recent commit - but, I'm not sure having a super long timeout by default is what we want to do? OTOH I don't have a firm reason why not I suppose.

Reply:
I have no strong opinion so long as it's always longer than server's by enough to let server return an empty response on its timeout

Reply:
We should treat start update as a long poll, hence the long timeout